cloudant-client/src/test/java/com/cloudant/tests/HttpTest.java

```diff
@@ -346,6 +346,70 @@ public void handleNonExpiry403() throws Exception {
         }
     }
 
+    /**
+     * Same as {@link #handleNonExpiry403()} but with no reason property in the JSON.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void handleNonExpiry403NoReason() throws Exception {
+
+        // Request sequence
+        // _session request to get Cookie
+        // GET request -> 403 (CouchDbException)
+        mockWebServer.enqueue(MockWebServerResources.OK_COOKIE);
+        mockWebServer.enqueue(new MockResponse().setResponseCode(403).setBody
+                ("{\"error\":\"403_not_expired_test\"}\r\n"));
+
+        try {
+            CloudantClient c = CloudantClientHelper.newMockWebServerClientBuilder(mockWebServer)
+                    .username("a")
+                    .password("b")
+                    .build();
+            //the GET request will try to get a session, then perform the GET
+            //the GET will result in a 403, which should result in a CouchDbException
+            c.executeRequest(Http.GET(c.getBaseUri()));
+            fail("A 403 not due to cookie expiry should result in a CouchDbException");
+        } catch (CouchDbException e) {
+            e.printStackTrace();
+            assertNotNull("The error should not be null", e.getError());
+            assertEquals("The error message should be the expected one", "403_not_expired_test", e
+                    .getError());
+        }
+    }
+
+    /**
+     * * Same as {@link #handleNonExpiry403()} but with a {@code null} reason property in the JSON.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void handleNonExpiry403NullReason() throws Exception {
```

Comment:
Done in bb18e34.
I also added the 403 renewal case it was also mostly the same.